src/derived/slot.rs

```diff
@@ -215,20 +218,27 @@ where
         // old value.
```

Comment:
Several lines above, `let database_key = self.database_key(db);` has only a single usage. Let's move that several lines up still, to reuse this local variable instead of two `self.database_key(db)` calls?